### ISSUE TITLE
An option not to load or update cover art data?

### ISSUE THREAD

**Issue by acroyear** [Zazama/node-id3#75]
Realizing why my reads for mp3s were so slow but writes were pretty quick.  There's a lot of data involved (and a lot of memory used up) when reading certain large tags that may not be needed.  Lyrics and Images in particular.  It would be great if we could have an option that would limit which tags to load into memory, so if all we're doing is organization based on artist/album/composer/conductor, or just cleaning up those tags, we're not wasting memory with the rest. (Of course, the update() call would need to have a similar option list so it doesn't accidentally wipe out tags just because it didn't load them).

I may propose an MR later in the week but I've a lot of other things on my plate (including my actual project i'm using this for :D ).

**Comment by acroyear** [comment]
Brainstorming: maybe providing an array of the tags to read/update, with pre-configured constants for the ones most used by certain software like iTunes and Subsonic.

**Comment by Zazama** [comment]
You're probably not gonna save that much performance by not loading certain tags, because the file still has to be searched (but not copied into a buffer).

![grafik](https://user-images.githubusercontent.com/7694808/91493740-07296000-e8b8-11ea-9250-cd4ecba817c3.png)

The top one is with APIC (picture) ignored, the bottom one is with picture loaded. There are probably not many cases where you're going to load a picture with more than two megabyte of size and at that size, there seems to be virtually no difference in loading time.

There'd be a memory optimization of course, maybe passing an option object with read like

```
NodeID3.read(file, { exclude: ['APIC'...] });
```

would be a good solution, where the exclusion process happens before line ~546.

**Comment by acroyear** [comment]
A follow-up on this. Some of the issue was using a debug version of nw.js, in order to get browser console access to ease with debugging.

This weekend I worked through the steps to actually package an executable (in mac case, a .app) in an optimized mode, and the loading and writing of mp3 files is exceptionally faster.  So while still a nice to have optimization if it might achieve something as far as memory usage goes, I'm no longer going "omg why is this so much slower than Java..." like I was at the time I filed this.

**Comment by Zazama** [comment]
Read part implemented in 0.2.0 by https://github.com/Zazama/node-id3/commit/aa46353174d759d85f298e4be971ed5192399d0a

Usage:
```javascript
// Possible options
const options = {
    include: ['TALB', 'TIT2'],    // only read the specified tags (default: all)
    exclude: ['APIC'],            // don't read the specified tags (default: [])
    onlyRaw: false,               // only return raw object (default: false)
    noRaw: false                  // don't generate raw object (default: false)
}
const tags = NodeID3.read(file, options)
```
